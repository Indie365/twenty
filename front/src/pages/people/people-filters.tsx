import { FilterDropdownCompanySearchSelect } from '@/companies/components/FilterDropdownCompanySearchSelect';
import { FilterConfigType } from '@/filters-and-sorts/interfaces/filters/interface';
import { EntityFilterWithField } from '@/filters-and-sorts/types/EntityFilterWithField';
import { SEARCH_COMPANY_QUERY } from '@/search/services/search';
import {
  IconBuildingSkyscraper,
  IconCalendarEvent,
  IconMail,
  IconMap,
  IconPhone,
  IconUser,
} from '@/ui/icons/index';
import { icon } from '@/ui/themes/icon';
import { Company, Person, QueryMode } from '~/generated/graphql';

export const peopleFilters: EntityFilterWithField<Person>[] = [
  {
    field: 'firstname',
    label: 'First name',
    icon: <IconUser size={icon.size.md} stroke={icon.stroke.sm} />,
    type: 'text',
  },
  {
    field: 'lastname',
    label: 'Last name',
    icon: <IconUser size={icon.size.md} stroke={icon.stroke.sm} />,
    type: 'text',
  },
  {
    field: 'email',
    label: 'Email',
    icon: <IconMail size={icon.size.md} stroke={icon.stroke.sm} />,
    type: 'text',
  },
  {
    field: 'companyId',
    label: 'Company',
    icon: (
      <IconBuildingSkyscraper size={icon.size.md} stroke={icon.stroke.sm} />
    ),
    type: 'entity',
    searchSelectComponent: <FilterDropdownCompanySearchSelect />,
  },
  {
    field: 'phone',
    label: 'Phone',
    icon: <IconPhone size={icon.size.md} stroke={icon.stroke.sm} />,
    type: 'text',
  },
  {
    field: 'createdAt',
    label: 'Created at',
    icon: <IconCalendarEvent size={icon.size.md} stroke={icon.stroke.sm} />,
    type: 'date',
  },
  {
    field: 'city',
    label: 'City',
    icon: <IconMap size={icon.size.md} stroke={icon.stroke.sm} />,
    type: 'text',
  },
];
