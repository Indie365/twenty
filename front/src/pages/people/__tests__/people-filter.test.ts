import { cityFilter, companyFilter } from '../people-filters';

describe('PeopleFilter', () => {
  it(`should render the filter ${companyFilter.key}`, () => {
    expect(
      companyFilter.operands[0].whereTemplate({
        id: 'test-id',
        name: 'test-name',
        domainName: 'test-domain-name',
        __typename: 'companies',
      }),
    ).toMatchSnapshot();
  });

  it(`should render the filter ${cityFilter.key}`, () => {
    expect(cityFilter.operands[0].whereTemplate('Paris')).toMatchSnapshot();
  });
});
