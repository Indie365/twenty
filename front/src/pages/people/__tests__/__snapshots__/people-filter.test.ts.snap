// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PeopleFilter City filter should generate the where variable of the GQL call 1`] = `
Object {
  "city": Object {
    "_eq": undefined,
  },
}
`;

exports[`PeopleFilter City filter should generate the where variable of the GQL call 2`] = `
Object {
  "_not": Object {
    "city": Object {
      "_eq": undefined,
    },
  },
}
`;

exports[`PeopleFilter Company fitler should generate the where variable of the GQL call 1`] = `
Object {
  "_and": Array [
    Object {
      "firstname": Object {
        "_eq": "undefined",
      },
    },
    Object {
      "lastname": Object {
        "_eq": "undefined",
      },
    },
  ],
}
`;

exports[`PeopleFilter Company fitler should generate the where variable of the GQL call 2`] = `
Object {
  "_not": Object {
    "_and": Array [
      Object {
        "firstname": Object {
          "_eq": "undefined",
        },
      },
      Object {
        "lastname": Object {
          "_eq": "undefined",
        },
      },
    ],
  },
}
`;

exports[`PeopleFilter Email filter should generate the where variable of the GQL call 1`] = `
Object {
  "email": Object {
    "_eq": undefined,
  },
}
`;

exports[`PeopleFilter Email filter should generate the where variable of the GQL call 2`] = `
Object {
  "_not": Object {
    "email": Object {
      "_eq": undefined,
    },
  },
}
`;

exports[`PeopleFilter Fullname filter should generate the where variable of the GQL call 1`] = `
Object {
  "_and": Array [
    Object {
      "firstname": Object {
        "_eq": "undefined",
      },
    },
    Object {
      "lastname": Object {
        "_eq": "undefined",
      },
    },
  ],
}
`;

exports[`PeopleFilter Fullname filter should generate the where variable of the GQL call 2`] = `
Object {
  "_not": Object {
    "_and": Array [
      Object {
        "firstname": Object {
          "_eq": "undefined",
        },
      },
      Object {
        "lastname": Object {
          "_eq": "undefined",
        },
      },
    ],
  },
}
`;
